### PR TITLE
[ADF-968] Added Site Dropdown component to document list

### PR DESCRIPTION
**Pull request by VitoAlbano**
**Please check if the PR fulfills these requirements**

> - [ ] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [] Bugfix
> - [X] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

**What is the new behaviour?**
A new dropdown which will fetch the sites of the user showing them , and for each selection an event is triggered 

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: ng2-components/ng2-alfresco-documentlist/README.md [hunk 1]
```diff
@@ -357,6 +360,29 @@ DocumentList provides simple breadcrumb element to indicate the current position
...
+
+| Name | Type | Description |
+| --- | --- | --- |
+| showDefaultOption | boolean | (optional) Shows into the dropdown a default option |
+
+### Events
+
+| Name | Returned Type | Description |
+| --- | --- | --- |
+| siteChanged | [SiteModel](https://github.com/Alfresco/alfresco-ng2-components/blob/development/ng2-components/ng2-alfresco-documentlist/src/models/site.model.ts) | emitted when user select a site  |
```
Comment (eromano) [inline ng2-components/ng2-alfresco-documentlist/README.md]: change as event could be enough

File: ng2-components/ng2-alfresco-documentlist/README.md [hunk 2]
```diff
@@ -357,6 +360,29 @@ DocumentList provides simple breadcrumb element to indicate the current position
 | --- | --- | --- |
 | navigate | [PathElementEntity](https://github.com/Alfresco/alfresco-js-api/blob/master/src/alfresco-core-rest-api/docs/PathElementEntity.md) |emitted when user clicks on a breadcrumb  |
 
+## Dropdown Site Component
+
+DocumentList provides simple dropdown element to indicate the current position within a navigation hierarchy.
```
Comment (eromano) [inline ng2-components/ng2-alfresco-documentlist/README.md]: +DocumentList provides simple dropdown element to indicate the current position within a navigation hierarchy I don't think is clearing explain what is doing.

File: ng2-components/ng2-alfresco-documentlist/README.md [hunk 3]
```diff
@@ -357,6 +360,29 @@ DocumentList provides simple breadcrumb element to indicate the current position
...
+  [showDefaultOption]="true"
+  (siteChanged)="getSiteContent($event)">
+ </adf-sites-dropdown>
+```
+
+### Properties
+
+| Name | Type | Description |
+| --- | --- | --- |
+| showDefaultOption | boolean | (optional) Shows into the dropdown a default option |
```
Comment (eromano) [inline ng2-components/ng2-alfresco-documentlist/README.md]: why do we need this property?

File: ng2-components/ng2-alfresco-core/src/models/site.model.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    guid: string;
+    description: string;
+    id: string;
+    preset: string;
+    title: string;
+    contents: SiteContentsModel[] = [];
+    members: SiteMembersModel[] = [];
+
+    constructor(obj?: any) {
+        this.role = obj && obj.entry && obj.entry.role;
```
Comment (eromano) [inline ng2-components/ng2-alfresco-core/src/models/site.model.ts]: can we check if obj and obj.entry are valid at the beginning just once?

File: ng2-components/ng2-alfresco-core/src/models/site.model.ts [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    role: string;
+    firstName: string;
+    emailNotificationsEnabled: boolean = false;
+    company: any;
+    id: string;
+    enable: boolean = false;
+    email: string;
+
+    constructor(obj?: any) {
+        this.role = obj && obj.role;
```
Comment (eromano) [inline ng2-components/ng2-alfresco-core/src/models/site.model.ts]: can we check if obj and is valid at the beginning just once?

File: demo-shell-ng2/app/components/sites/sites-demo.component.css [hunk 6]
```diff
@@ -0,0 +1,7 @@
+
+.demo-site-container-style {
```
Comment (eromano) [inline demo-shell-ng2/app/components/sites/sites-demo.component.css]: can we use the adf prefix?

File: demo-shell-ng2/app/components/sites/sites-demo.component.html [hunk 7]
```diff
@@ -0,0 +1,12 @@
+<div class="demo-site-container-style" id="demo-container">
```
Comment (eromano) [inline demo-shell-ng2/app/components/sites/sites-demo.component.html]: why do we need a new page can not show this functionality in the current files.component?

File: ng2-components/ng2-alfresco-documentlist/src/components/site-dropdown/sites-dropdown.component.css [hunk 8]
```diff
@@ -0,0 +1,3 @@
+.adf-site-list-element {
```
Comment (eromano) [inline ng2-components/ng2-alfresco-documentlist/src/components/site-dropdown/sites-dropdown.component.css]: can you make it scss?

File: ng2-components/ng2-alfresco-documentlist/README.md (hunk 3 above)
Comment (VitoAlbano) [inline ng2-components/ng2-alfresco-documentlist/README.md]: To enable/disable from outside the default option in a select. Sometimes you want a default one which has no value associated.

File: demo-shell-ng2/app/components/sites/sites-demo.component.html (hunk 7 above)
Comment (VitoAlbano) [inline demo-shell-ng2/app/components/sites/sites-demo.component.html]: I'll integrate it there.

File: demo-shell-ng2/app/components/sites/sites-demo.component.css (hunk 6 above)
Comment (VitoAlbano) [inline demo-shell-ng2/app/components/sites/sites-demo.component.css]: done

File: demo-shell-ng2/app/components/files/files.component.html [hunk 9]
```diff
@@ -48,29 +45,13 @@
...
-                [creationMenuActions]="!useCustomToolbar"
-                [currentFolderId]="currentFolderId"
-                [contextMenuActions]="true"
-                [contentActions]="true"
-                [allowDropFiles]="true"
-                (error)="onNavigationError($event)"
-                (success)="resetError()"
-                (preview)="showFile($event)"
-                (permissionError)="handlePermissionError($event)">
+        <adf-document-list #documentList [permissionsStyle]="permissionsStyle" [creationMenuActions]="!useCustomToolbar" [currentFolderId]="currentFolderId"
```
Comment (DenisVuyka) [inline demo-shell-ng2/app/components/files/files.component.html]: Can we get back the multiline approach for the attributes to have better readability?

Comment (VitoAlbano) [inline demo-shell-ng2/app/components/files/files.component.html]: yes sure

**Pull request merged**